Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrash.m

```diff
@@ -345,32 +345,38 @@ - (void)deleteAllReports {
     [self.crashReportStore deleteAllFiles];
 }
 
-- (void)reportUserException:(NSString *)name
-                     reason:(NSString *)reason
-                   language:(NSString *)language
-                 lineOfCode:(NSString *)lineOfCode
-                 stackTrace:(NSArray *)stackTrace
-           terminateProgram:(BOOL)terminateProgram {
-    const char *cName = [name cStringUsingEncoding:NSUTF8StringEncoding];
-    const char *cReason = [reason cStringUsingEncoding:NSUTF8StringEncoding];
-    const char *cLanguage =
-        [language cStringUsingEncoding:NSUTF8StringEncoding];
-    const char *cLineOfCode =
-        [lineOfCode cStringUsingEncoding:NSUTF8StringEncoding];
+- (const char *)encodeAsJSONString:(id)object {
```

Comment:
It may be worth moving this out to a different source file, as this method doesn't feel exclusively related to creating crash reports, and it'd be good to have this level of error handling anywhere else we're serialising JSON. It would also make it easier to add unit test coverage

Reply:
The method is used in only one place, and is private to this file. I'll move it to the utility function section to make the intent clearer.